### ISSUE TITLE
any basic generators?

### ISSUE THREAD

**Issue by jmagaram** [seangenabe/starry#23]
Does the library have any basic generators, like "the numbers from 1 to 1000"? This would be a nice addition. Check out [F# Seq module](https://msdn.microsoft.com/en-us/visualfsharpdocs/conceptual/collections.seq-module-%5Bfsharp%5D). Some basic generators like...

unfold
init
initInfinite

...would go a long way. Here is an unfold implementation...

    export function unfold<T, TState>(args: { seed: TState, generator: (state: TState) => ([T, TState] | undefined) }) {
        function* items() {
            let state: TState | undefined = args.seed;
            do {
                const next = args.generator(state);
                if (next !== undefined) {
                    yield next[0];
                    state = next[1];
                }
                else {
                    state = undefined;
                }
            } while (state !== undefined)
        }
        return {
            [Symbol.iterator]: items
        }
    }

**Comment by jmagaram** [comment]
Actually I’m realizing now how easy it is to create an iterable using the function* syntax. So maybe there is not much need for unfold, range, init, repeat, and other ways of creating an iterable in your library.

**Comment by seangenabe** [comment]
Wouldn't `unfold` just be equivalent to a `map` function?

I think `init`/`initInfinite` would be covered with a `function*` generator function, yes.

Right now I'm just adding modules as I need them, but I'm open to ideas. I'm taking the .NET Framework and lodash as inspirations.

**Comment by jmagaram** [comment]
No. Unfold is like the inverse of fold. Take an initial state and from that generate a sequence, each time generating a new item to be emitted and the next state. Probably can do that with custom generators. Some good inspiration here too.

https://msdn.microsoft.com/en-us/visualfsharpdocs/conceptual/collections.seq-module-%5Bfsharp%5D




________________________________
